### DISCUSSION TITLE
[Q&A] Deploy Update GLPI via Deploy inventory

### DISCUSSION THREAD

**Discussion by franga92** [glpi-project/glpi-agent#666] [Q&A]
### Documentation & bug reporting acknowledgment

Yes, I read it

### Describe your problem

Hello, everyone,

I was wondering if it is possible to install the agent update via Task.

Unfortunately I'm left with several machines in DMZ and linux machines that I have no other way but to upgrade them manually and automating the upgrade would be more convenient in terms of time. is it possible to do what I'm asking?

**Comment by g-bougard** [comment]
Hi @franga92 

this is probably possible, but to be honest, I say "probably" because I never tried myself.

In fact, in place of upgrade the agent, you should plan an upgrade of the agent. To do so, you have in the job script to copy installer and eventually scripts to a dedicated place and manage to planify the upgrade later. The reason is you should not upgrade the agent at the same time you're running a task or you take the risk the task is itself cancelled, leaving the agent in an unknown state.

On linux, for example, you can create an upgrade script which will just start the installer with your required arguments. Then in the job script, you can copy if to /etc/cron.dayly and take care it is executable with chmod command. Also I think you can simply copy the installer itself so it will just upgrade the agent with the current configuration.

You also probably want to remove the copied script after the upgrade has been done. Or you can leave it: the script will be started but the upgrade will leave has the version is still installed. But you'll have to remove it if you want to upgrade to a newer version.

Note this: it is important to plan the upgrade at a different hour than the job planned time. Indeed few minutes can be sufficient, but taking at least an hour is just by security to be sure you won't stop other important task.

If you have some script development skills, you can also create a script which will monitor the service status and start it in the background from the job script, maybe with screen command. To monitor the agent status and if you have leave enabled httpd interface and put 127.0.0.1, a request to `http://127.0.0.1:62354/` will give you a simple line of text like `status: running Deploy` or `status: waiting`. At the moment you see `waiting` in that string, you know you can upgrade the agent safely.

On windows, if you have also this target in mind, you can use schtasks command to planify the start of a bat which will run the install as you expect in silent mode.
Again you should remove the planned task after the upgrade.

I hope this helps.
